### PR TITLE
ddl: Fix error code bug and update ddl test

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: ddl/ddl_worker_test.go [hunk 1]
```diff
@@ -335,31 +220,31 @@ func (s *testDDLSuite) TestColumnError(c *C) {
...
-		Type:     model.ActionDropColumn,
-		Args:     []interface{}{1},
-	}
+	return job
+}
 
-	err = d.doDDLJob(ctx, job)
-	c.Assert(err, NotNil)
-	testCheckJobCancelled(c, d, job)
+func (s *testDDLSuite) TestSyncMode(c *C) {
```
Comment (coocood) [inline ddl/ddl_worker_test.go]: Why define an empty Test?

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**